--- ../llvm_android_origin/toolchains.py	2021-11-09 18:35:36.572648528 +0800
+++ toolchains.py	2021-11-10 01:49:55.671911164 +0800
@@ -115,7 +115,7 @@ class Toolchain:
     @property
     def lib_dir(self) -> Path:
         """Returns the path to lib dir."""
-        return self.path / 'lib64'
+        return self.path / 'lib'
 
     @property
     def _version_file(self) -> Path:
@@ -139,7 +139,7 @@ class Toolchain:
 
     @property
     def libcxx_headers(self) -> Path:
-        return self.path / 'include' / 'c++' / 'v1'
+        return self.path / 'include' / 'c++' / '10.2.0'
 
 
 def get_prebuilt_toolchain() -> Toolchain:
